xslt/functions.go

```diff
@@ -210,3 +215,48 @@ func EXSLTnodeset(context xpath.VariableScope, args []interface{}) interface{} {
 
 	return nodes
 }
+
+func EXSLTmathconstant(context xpath.VariableScope, args []interface{}) interface{} {
+
+	if len(args) != 2 {
+		return 0
+	}
+
+	name := args[0]
+	precision := int(args[1].(float64))
+
+	switch name {
+	case "PI":
+		return fmt.Sprintf("%.*f", precision, 3.1415926535897932384626433832795028841971693993751)
+	case "E":
+		return fmt.Sprintf("%.*f", precision, 2.71828182845904523536028747135266249775724709369996)
+	case "SQRRT2":
+		return fmt.Sprintf("%.*f", precision, 1.41421356237309504880168872420969807856967187537694)
+	case "LN2":
+		return fmt.Sprintf("%.*f", precision, 0.69314718055994530941723212145817656807550013436025)
+	case "LN10":
+		return fmt.Sprintf("%.*f", precision, 2.30258509299404568402)
+	case "LOG2E":
+		return fmt.Sprintf("%.*f", precision, 1.4426950408889634074)
+	case "SQRT1_2":
+		return fmt.Sprintf("%.*f", precision, 0.70710678118654752440)
+	}
```

Comment:
There should be a default that outputs an error message (in case someone passes an unsupported constant).